src/pdn/src/straps.cpp

```diff
@@ -404,12 +459,40 @@ void FollowPins::makeShapes(const Shape::ShapeTreeMap& other_shapes)
     auto* power_strap = new FollowPinShape(
         layer, power, odb::Rect(x0, power_y_bot, x1, power_y_bot + width));
     power_strap->addRow(row);
-    addShape(power_strap);
+    if (getExtendMode() == CLIP_CELLS) {
+      std::vector<Shape*> replacements;
+      if (power_strap->cut(power_obs_tree, grid, replacements)) {
+        for (Shape *cut_shape : replacements) {
+          if (cut_shape->getRect().dx() > 20 * width) {
+            cut_shape->setLocked();
+            addShape(cut_shape);
+          }
+        }
+      } else {
+        addShape(power_strap);
+      }
+    } else {
+      addShape(power_strap);
+    }
```

Comment:
this is pretty ugly, it might be cleaner to have the `FollowPins` class override the `addShape` to make this a little less messy.